srcpkgs/nethack/patches/1-linux-hints.patch

```diff
@@ -0,0 +1,19 @@
+Undefining SECURE is needed so that the nethack binary does not
+drop privileges when run with the SUID bit.
```

Comment:
I don't understand exactly what you mean by this. Why do you want it to not drop privileges when run with SUID? (Is it installed as SUID?)

For a change like this, I think it's important to explain why you're making it, not only what it does.

Reply:
Yes, it took me a while to find and understand why this is needed. I will try to explain this in a separate message, but for now let me just point out that by applying this patch we are not actually _changing_ anything, but maintaining the previous behavior. 

Revision 3.6.6_1 of the package does not define `SECURE`, in fact, it overrides all `CFLAGS` set in `sys/unix/hints/linux` and re-defines only a subset of them as an argument to `make all`. When moving away from this approach (and thus applying all the `CFLAGS` from the hints file) I discovered that defining `SECURE` in our setup would break the package entirely.

Reply:
Ah! Indeed, I had forgotten that we completely overrode the CFLAGS. Thanks for reminding me about it :)